js/reference.js

```diff
@@ -3062,7 +3062,13 @@
 
                             // in entry mode, pseudo-column, inbound fk, and key are not allowed
                             if (!(isEntry && (refCol.isPathColumn || refCol.isInboundForeignKey || refCol.isKey) )) {
-                                this._referenceColumns.push(refCol);
+                                // NOTE: should I check refCol.hasAggregate ?
+                                // check we are in entry/compact mode before excluding more columns
+                                if (context.startsWith(module._contexts.COMPACT_ENTRY) && (refCol.hasWaitFor || !refCol.isUnique || (refCol.hasPath && refCol.isUnique && refCol.foreignKeys.length > 1)) ) {
```

Comment:
To make it a little bit "cleaner", you could do something like

```javascript
ignore = !(isEntry && (refCol.isPathColumn || refCol.isInboundForeignKey || refCol.isKey) ) ||
         NEW_CONDITION;
if (!ignore) {
  this._referenceColumns.push(refCol);
}
```

And as for the new condition, you can make it simpler. What I wrote originally was a literal translation of what we want but I understand that it's not easy to read and it's just too long.

- Technically since it's non-detailed, you could just check for `hasAggregate`.
- We don't need to check all that three conditions and just `refCol.isPathColumn && refCol.foreignKeys.length > 1` should be enough.

Also, this function works with "uncontextualized" references and let's just make sure it still continues to do so

So 
```javascript
// this should be added at the beginning of this function where we're initializing all these type of variabels
var isCompactEntry = typeof context === "string" && context.startsWith(module._contexts.COMPACT_ENTRY);

...


ignore = (isEntry && (refCol.isPathColumn || refCol.isInboundForeignKey || refCol.isKey) ) ||
         (isCompactEntry && (refCol.hasWaitFor || refCol.hasAggregate || (refCol.isPathColumn && refCol.foreignKeys.length > 1));
```